packages/x-components/src/x-modules/url/store/actions/update-url.action.ts

```diff
@@ -1,61 +1,107 @@
+import { ActionsClass } from '../../../../store/actions.types';
 import { Dictionary } from '../../../../utils';
-import { forEach, reduce } from '../../../../utils/object';
-import { UrlParamValue, UrlXStoreModule } from '../types';
+import { forEach } from '../../../../utils/object';
+import { UrlActionContext, UrlParamKey, UrlParamValue, UrlXStoreModule } from '../types';
 
 /**
- * Default implementation for the {@link UrlActions.updateUrl}.
+ * Class implementation for the {@link UpdateUrlAction.updateUrl} action.
  *
- * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
- * provided by Vuex.
- *
- * @internal
+ * @public
  */
-export const updateUrl: UrlXStoreModule['actions']['updateUrl'] = ({
-  getters: { urlParams, urlMappedParamNames }
-}) => {
-  const url = new URL(window.location.href);
-  const oldParams = url.searchParams;
+export class UpdateUrlAction implements ActionsClass<UrlXStoreModule> {
+  /** Parameters that will require to push the browser state instead of replace. */
```

Comment:
Why this is a class variable? It's only used inside the `pushableParamsChanged`.

Reply:
I followed the same approach we did with the `urlParams` getter. Moving this to a variable to make it more easy to find when we make changes if needed.

Reply:
The idea of having a variable is really good, what I meant was to declare inside the method `pushableParamsChanged` as it's the only place where it0s used.

Reply:
This way you don't have to look for the variable into a function.